text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,718 @@
+   * @default no vpcs are associated
+   */
+  readonly vpcs?: ec2.IVpc[] | undefined;
+  /**
+   * Allow external principals
+   * @default false
+   */
+  readonly accessmode?: ServiceNetworkAccessMode | undefined;
+}
+```
+
+### Service
+
+A service within VPC Lattice is an independently deployable unit of software that delivers a specific task or function.  A service has listeners that use rules, that you configure to route traffic to your targets. Targets can be EC2 instances, IP addresses, serverless Lambda functions, Application Load Balancers, or Kubernetes Pods.  The following diagram shows the key components of a typical service within VPC Lattice.
+
+
+![service](https://docs.aws.amazon.com/images/vpc-lattice/latest/ug/images/service.png)
+
+
+```typescript
+/**
+ * Create a vpcLattice service network.
+ * Implemented by `Service`.
+ */
+export interface IService extends core.IResource {
+  /**
+  * The Amazon Resource Name (ARN) of the service.
+  */
+  readonly serviceArn: string;
+  /**
+  * The Id of the Service Network
+  */
+  readonly serviceId: string;
+  /**
+   * Allow an Odd
+   */
+  readonly orgId: string | undefined;
+  /**
+   * the policy document for the service;
+   */
+  authPolicy: iam.PolicyDocument;
+
+  /**
+   * Add A vpc listener to the Service.
+   * @param props
+   */
+  shareToAccounts(props: ShareServiceProps): void;
+
+  /**
+   * Grant Access to other principals
+   */
+  grantAccess(principals: iam.IPrincipal[]): void;
+
+  /**
+   * Apply the authAuthPolicy to the Service
+   */
+  applyAuthPolicy(): iam.PolicyDocument;
+  /**
+   * Add A policyStatement to the Auth Policy
+   */
+  addPolicyStatement(statement: iam.PolicyStatement): void;
+}
+export interface IService extends core.IResource {
+  /**
+  * The Amazon Resource Name (ARN) of the service.
+  */
+  readonly serviceArn: string;
+  /**
+  * The Id of the Service Network
+  */
+  readonly serviceId: string;
+  /**
+   * Allow an Odd
+   */
+  readonly orgId: string | undefined;
+  /**
+   * the policy document for the service;
+   */
+  authPolicy: iam.PolicyDocument;
+
+  /**
+   * Add A vpc listener to the Service.
+   * @param props
+   */
+  shareToAccounts(props: ShareServiceProps): void;
+
+  /**
+   * Grant Access to other principals
+   */
+  grantAccess(principals: iam.IPrincipal[]): void;
+
+  /**
+   * Apply the authAuthPolicy to the Service
+   */
+  applyAuthPolicy(): iam.PolicyDocument;
+  /**
+   * Add A policyStatement to the Auth Policy
+   */
+  addPolicyStatement(statement: iam.PolicyStatement): void;
+}
+
+```
+
+`Service` will take `ServiceProps` as props
+```typescript
+/**
+ * Properties for a Lattice Service
+ */
+export interface LatticeServiceProps {
+
+  /**
+   * Name for the service
+   * @default cloudformation will provide a name
+   */
+  readonly name?: string | undefined;
+
+  /**
+   * The authType of the Service
+   * @default 'AWS_IAM'
+   */
+  readonly authType?: string | undefined;
```

Comment:
This was an enum elsewhere, which I assume was also the intention here? Could you please update this to make it consistent?

Reply:
Good Spotting. that authType should be refering to the enum as well.